daisy/workflow/workflow.go

```diff
@@ -223,20 +238,52 @@ func (w *Workflow) populateStep(step *Step) error {
 	step.SubWorkflow.workflow.StorageClient = w.StorageClient
 	step.SubWorkflow.workflow.Ctx = w.Ctx
 	step.SubWorkflow.workflow.Cancel = w.Cancel
-	step.SubWorkflow.workflow.logger = w.logger
+	step.SubWorkflow.workflow.gcsLogWriter = w.gcsLogWriter
+	step.SubWorkflow.workflow.vars = map[string]vars{}
 	for k, v := range step.SubWorkflow.Vars {
-		step.SubWorkflow.workflow.Vars[k] = v
+		step.SubWorkflow.workflow.vars[k] = vars{Value: v}
 	}
 	return step.SubWorkflow.workflow.populate()
 }
 
+func (w *Workflow) populateVars() error {
+	if w.vars == nil {
+		w.vars = map[string]vars{}
+	}
+	for k, v := range w.Vars {
+		// Don't overwrite existing vars (applies to subworkflows).
+		if _, ok := w.vars[k]; ok {
+			continue
+		}
+		var sv string
+		if err := json.Unmarshal(v, &sv); err == nil {
+			w.vars[k] = vars{Value: sv}
+			continue
+		}
+		var vv vars
+		if err := json.Unmarshal(v, &vv); err == nil {
+			if vv.Required && vv.Value == "" {
+				return fmt.Errorf("required var %q cannot be blank", k)
+			}
+			w.vars[k] = vv
+			continue
+		}
+		return fmt.Errorf("cannot unmarshal Var %q, value: %s", k, v)
+	}
+	return nil
+}
+
 func (w *Workflow) populate() error {
```

Comment:
I've been trying to parse the changes here. Is anything going to change with how autovars vs vars get resolved?